src/game/assets.rs

```diff
@@ -4,38 +4,44 @@ use bevy::{
     utils::HashMap,
 };
 
+pub(super) fn plugin(app: &mut App) {
+    app.register_type::<(
+        AssetMap<ImageKey>,
+        AssetMap<SfxKey>,
+        AssetMap<SoundtrackKey>,
+    )>();
+    app.add_systems(Startup, preload_assets);
+}
+
+fn preload_assets(mut commands: Commands, asset_server: Res<AssetServer>) {
+    commands.insert_resource(ImageKey::load(&asset_server));
```

Comment:
Implement `FromWorld` for all `T: AssetKey + Resource` and these can be simple `init_resource` calls.

Reply:
It'd be `FromWorld` for `AssetMap<K>`, and this would become `app.init_resource::<AssetMap<ImageKey>>()`. Which is reasonable.